controllers/pkg/reconcilers/bootstrap-packages/reconciler.go

```diff
@@ -89,144 +82,135 @@ func (r *reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, c a
 type reconciler struct {
 	client.Client
 	porchClient client.Client
-
-	l logr.Logger
+	log         logr.Logger
 }
 
 func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.l = log.FromContext(ctx)
-	cr := &porchv1alpha1.PackageRevision{}
-	if err := r.Get(ctx, req.NamespacedName, cr); err != nil {
+	r.log = log.FromContext(ctx)
+	packageRevision := &porchv1alpha1.PackageRevision{}
+	if err := r.Client.Get(ctx, req.NamespacedName, packageRevision); err != nil {
 		// There's no need to requeue if we no longer exist. Otherwise we'll be
 		// requeued implicitly because we return an error.
 		if resource.IgnoreNotFound(err) != nil {
 			msg := "cannot get resource"
-			r.l.Error(err, msg)
+			r.log.Error(err, msg)
 			return ctrl.Result{}, errors.Wrap(resource.IgnoreNotFound(err), msg)
 		}
 		return ctrl.Result{}, nil
 	}
 
 	// check if the packagerevision is part of a staging repository
 	// if not we can ignore this package revision
-	stagingPR, err := r.IsStagingPackageRevision(ctx, cr)
+	stagingPR, err := r.IsStagingPackageRevision(ctx, r.porchClient, packageRevision.Spec.RepositoryName)
```

Comment:
it is not good practice to add a variable that is already in your context also as a parameter.

Reply:
I will admit, I struggled to unit test these functions. Passing it as a parameter was the obvious way to allow me to mock the client. If there is an alternative way, without passing it in, please do advise when you have time.
I looked at trying to use the fake.go but because I needed to mock actual return values, I didn't get very far with it.

Reply:
you can create a new reconciler in which you pass this info. So build a reconciler in the Mock that has the Mock client and you call the fn like that.

Reply:
Done